packages/vite-plugin-nitro/src/lib/build-sitemap.ts

```diff
@@ -0,0 +1,73 @@
+import { SitemapConfig } from './options';
+import * as fs from 'fs';
+import { XMLBuilder } from 'xmlbuilder2/lib/interfaces';
+import { create } from 'xmlbuilder2';
+import { UserConfig } from 'vite';
+import * as path from 'path';
+
+export type PagesJson = {
+  page: string;
+  lastMod: string;
+};
+
+export async function buildSitemap(
+  config: UserConfig,
+  sitemapConfig: SitemapConfig,
+  routes: string[] | (() => Promise<(string | undefined)[]>)
+) {
+  const routeList: string[] = await optionHasRoutes(routes);
+
+  if (routeList.length) {
+    const sitemapData: PagesJson[] = routeList.map((page: string) => ({
+      page: `https://www.${sitemapConfig.domain}${page.replace(/^\/+/g, '')}`,
```

Comment:
Let the user customize the entire host, for example `https://myblog.dev`

```suggestion
      page: `${sitemapConfig.host}${page.replace(/^\/+/g, '')}`,
```